llvm/lib/Target/AMDGPU/AMDGPUAsmPrinter.cpp

```diff
@@ -1025,6 +1025,26 @@ void AMDGPUAsmPrinter::EmitProgramInfoSI(const MachineFunction &MF,
   OutStreamer->emitInt32(MFI->getNumSpilledVGPRs());
 }
 
+// Helper function to add common PAL Metadata 3.0+
+static void EmitPALMetadataCommon(AMDGPUPALMetadata *MD,
+                                  const SIProgramInfo &CurrentProgramInfo,
+                                  CallingConv::ID CC,
+                                  const MCSubtargetInfo &ST) {
+  MD->setHwStage(CC, ".ieee_mode", (bool)CurrentProgramInfo.IEEEMode);
```

Comment:
should this be skipped for gfx12?

Reply:
Maybe, but I wasn't sure. Seems we generally do not bother to do gfxip checks for others (e.g., "MemOrdered" is only supported on gfx10 and gfx11).

Reply:
I can easily add that though - and that does mirror the recent change to getPGMRsrc1.

Reply:
Sounds good.